### PR TITLE
MacOS_support

### PR DESCRIPTION
**Pull request by Nunez2196**
this adds the OS specific code needed for Mac releases

### CODE REVIEW

File: src/unpack.js [hunk 1]
```diff
@@ -64,6 +64,23 @@ function unpackRelease(unpackFolder, targetFolderName, archiveFile, progressElem
...
             }
+        } else if(os.platform() === "darwin"){
+            // Find from PATH
+            unpacker = which.sync("7za", {nothrow: true});
+
+            if(!unpacker){
+                // Use packed in version //
+                console.log("No 7za found in PATH, using packed in one");
+
+                unpacker = path.join(remote.app.getAppPath(), "tools/7zip/7za");
```
Comment (hhyyrylainen) [inline src/unpack.js:75]: Does this actually work? Isn't this the linux executable?

Comment (Nunez2196) [reply src/unpack.js:75]: Actually no, let me remove that.

File: src/unpack.js [hunk 2]
```diff
@@ -64,6 +64,25 @@ function unpackRelease(unpackFolder, targetFolderName, archiveFile, progressElem
...
+            // Find from PATH
+            unpacker = which.sync("7za", {nothrow: true});
+
+            if(!unpacker){
+                // Some users on Mac import Linux versions of 7zip through homebrew //
+                console.log("No 7za found in PATH, using packed in one");
+
+                // Popular 7zip alternative for Mac users, linked on 7zip page at line 81 //
+                unpacker = path.join(remote.app.getAppPath(),
+                    "tools/7zip/Keka.app/Contents/MacOS/keka7z");
```
Comment (hhyyrylainen) [inline src/unpack.js:77]: You refer to this, but you didn't include the actual executable?
Why not use the version from homebrew? At work I've found homebrew to be one of the things that  makes using mac tolerable.

File: src/unpack.js [hunk 3]
```diff
@@ -64,6 +64,24 @@ function unpackRelease(unpackFolder, targetFolderName, archiveFile, progressElem
...
+        } else if(os.platform() === "darwin"){
+            // Find from PATH
+            unpacker = which.sync("7za", {nothrow: true});
+
+            if(!unpacker){
+                // Use packed in version for mac //
+                console.log("No 7za found in PATH, using packed in one");
+
+                unpacker = path.join(remote.app.getAppPath(),
+                    "tools/7zip/7za(mac)");
```
Comment (hhyyrylainen) [inline src/unpack.js:76]: This doesn't match the actual path.

File: src/unpack.js [hunk 4]
```diff
@@ -141,6 +159,9 @@ function getThriveExecutableName(){
     } else if(os.platform() === "linux"){
 
         return "Thrive";
+    } else if(os.platform() === "darwin"){
+
+        return "Thrive.app/Contents/MacOS/Thrive";
```
Comment (hhyyrylainen) [inline src/unpack.js:164]: Now that I had a good look, this is not consistent with the other platforms. This should return "Thrive" and the function that calls this, which is responsible for finding the executable path, should be updated instead to properly find the executable on mac.

Comment (Nunez2196) [reply src/unpack.js:164]: you mean the function below it: thriveExecutableExistsInFolder(folderToCheck) ?

Comment (hhyyrylainen) [reply src/unpack.js:164]: I mean `findBinInRelease`, which when properly updated for mac support should make the entire process of finding the executable work: https://github.com/Revolutionary-Games/Thrive-Launcher/blob/4c1eeb03108e1ec3e3f3f347376f4e77f65728fc/src/thrive_runner.js#L17
the reason that it isn't a hardcoded path is that the thrive folder structure has changed multiple times in the past.

File: src/unpack.js [hunk 5]
```diff
@@ -138,10 +156,11 @@ function getThriveExecutableName(){
 
         return "Thrive.exe";
 
-    } else if(os.platform() === "linux"){
+    } else if(os.platform() === "linux" || "darwin"){
```
Comment (hhyyrylainen) [inline src/unpack.js:159]: This line probably doesn't do the intended thing. Looks like the linter caught this.

Comment (Nunez2196) [reply src/unpack.js:159]: should I just make Mac have it's own conditional again then? linux and mac call the .exe the same thing.

Comment (hhyyrylainen) [reply src/unpack.js:159]: JS doesn't support checking multiple conditions like that...
You need to write out the entire thing: `os.platform() === "linux" || os.platform() === "darwin"`

**Review by hhyyrylainen — APPROVED** [review]
Assuming that this actually works on mac, this looks good to me.

**Pull request merged**